pkg/build/builder/dockerutil.go

```diff
@@ -156,40 +157,43 @@ func dockerRun(client DockerClient, createOpts docker.CreateContainerOptions, lo
 
 	containerName := containerNameOrID(c)
 
-	// Container was created, so we defer its removal.
-	defer func() {
+	// Container was created, so we defer its removal, and also remove it if we get a sigterm
+
+	removeContainer := func() {
 		glog.V(4).Infof("Removing container %q ...", containerName)
 		if err := client.RemoveContainer(docker.RemoveContainerOptions{ID: c.ID}); err != nil {
 			glog.Warningf("Failed to remove container %q: %v", containerName, err)
 		} else {
 			glog.V(4).Infof("Removed container %q", containerName)
 		}
-	}()
-
-	// Start the container.
-	glog.V(4).Infof("Starting container %q ...", containerName)
-	if err := client.StartContainer(c.ID, nil); err != nil {
-		return fmt.Errorf("start container %q: %v", containerName, err)
 	}
 
-	// Stream container logs.
-	logsOpts.Container = c.ID
-	glog.V(4).Infof("Streaming logs of container %q with options %+v ...", containerName, logsOpts)
-	if err := client.Logs(logsOpts); err != nil {
-		return fmt.Errorf("streaming logs of %q: %v", containerName, err)
-	}
+	return interrupt.New(nil, removeContainer).Run(func() error {
```

Comment:
i'm happy to make it a named function (i've asked others to do the same before), but i can't make it a first class function w/ args because the InterruptHandler isn't designed to take functions w/ args.  so it needs to be a closure.